Add single psr output

SimonStevenson opened the pull request:
Add SSE PSR Output to resolve issue https://github.com/TeamCOMPAS/COMPAS/issues/1333 . See that issue for further context and details. 

Current outstanding issue to be resolved before merging: Output appears with name BSE_Pulsar_Evolution rather than SSE_Pulsar_Evolution. I called it SSE_Pulsar_Evolution in `LogTypedefs.h`. I assume it is picking up the name from the `--logfile-pulsar-evolution` option (which defaults to BSE_Pulsar_Evolution), but the same is true for SSE_Supernovae (default value for `--logfile-supernovae` is BSE_Supernovae) and that correctly shows up as SSE_Supernovae. I assume I just missed something somewhere. [EDIT: now solved. Indeed, I missed something in `Options.h`]

Example test line:
```
./COMPAS --mode SSE -n 1 --initial-mass 10.0 --pulsar-magnetic-field-decay-timescale 1000.0 --evolve-pulsars TRUE --random-seed 0 --maximum-evolution-time 1000.0 --pulsar-birth-magnetic-field-distribution LOGNORMAL --pulsar-birth-spin-period-distribution NORMAL
```

Some example test plots:
![single_psr_test_B_v_t](https://github.com/user-attachments/assets/e61dbf88-862a-48dd-9061-dcd858fa97bb)
![single_psr_test_f_v_t](https://github.com/user-attachments/assets/3e5b180e-9b33-4701-8040-fcb613c95ede)

SimonStevenson committed:
Adding SSE_Pulsar_Evolution output file

SimonStevenson committed:
Editing stopping condition to allow neutron stars to continue evolving if EvolvePulsars is true

SimonStevenson committed:
Added a call to SpinDownIsolatedPulsar to Star::EvolveOneTimestep to update pulsar attributes

SimonStevenson committed:
Updating documentation

SimonStevenson commented:
Also realised I need to add something to the What's New page, will do that now

SimonStevenson committed:
Add an entry to Whats New page

SimonStevenson commented:
Figured out the name issue

SimonStevenson committed:
Fixed SSE_ v BSE_ prefix issue, change in Options.h

ilyamandel reviewed (approved):
Hi @SimonStevenson ,
Looks good, happy to approve.  Two things for your consideration:

1. Star::EvolveOneTimeStep

I found the following a bit hard to parse.  

else if (m_Star->IsOneOf(COMPACT_OBJECTS) && !OPTIONS->EvolvePulsars()){                            // If star is a compact object and we are not evolving pulsars
                evolutionStatus = EVOLUTION_STATUS::DONE;                                                       // yes - we're done	                evolutionStatus = EVOLUTION_STATUS::DONE;                                                       // we are done
            }
            else if (m_Star->IsOneOf(WHITE_DWARFS) || m_Star->StellarType() == STELLAR_TYPE::BLACK_HOLE){       // If compact object is a WD or BH, we're done even if we are evolving pulsars
                evolutionStatus = EVOLUTION_STATUS::DONE;
            }	           
            
            
What if you instead just do this?

 else if (m_Star->IsOneOf(WHITE_DWARFS) || m_Star->StellarType() == STELLAR_TYPE::BLACK_HOLE ||  (m_Star->StellarType() == STELLAR_TYPE::NEUTRON_STAR && !OPTIONS->EvolvePulsars() ) ){       // If a WD or BH, or an NS but not evolving pulsars, we're done 
                evolutionStatus = EVOLUTION_STATUS::DONE;
            }	            
            
            
2. LogTypedefs.h

The list seems to be in lexicographic order, except for this new addition -- so maybe keep it ordered?

    SSE_DETAILED_OUTPUT,
    SSE_SUPERNOVAE,
    SSE_SWITCH_LOG,	    
    SSE_SYSTEM_PARAMETERS,
    SSE_PULSAR_EVOLUTION

SimonStevenson committed:
Reordering in LogTypedefs.h

SimonStevenson committed:
Implementing Ilya's review suggestions to simplify stopping condition in Star.cpp

SimonStevenson commented:
Thanks for the review Ilya!

1. Thanks, I agree that's a bit simpler than what I had.
2. Fixed!  

I've implemented and tested these changes. Will now merge.

ilyamandel reviewed (approved):
Thanks, @SimonStevenson !
